recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -0,0 +1,496 @@
+    }, {
+        "require": "zlib/1.2.12",
+        "default": True,
+        "dep": "ZLIB",
+        "descr": "zlib (external)"
+    }, {
+        "conan_dep": "zstd::zstdlib",
+        "require": "zstd/1.5.2",
+        "dep": "ZSTD",
+        "descr": "ZSTD compression library"
+    }, {
+        "dep": "rdb",
+        "descr": "enable RIEGL RDB library"
+    },
+    {
+        "conan_dep": [ "libkml::kmldom", "libkml::kmlengine" ],
+        "dep": "LibKML",
+        "require": "libkml/1.3.0"
+    }]
+
+    options = _make_options(gdal_deps)
+    default_options = _make_default_options(gdal_deps)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.with_sqlite3:
+            self.options["sqlite3"].enable_column_metadata = True
+
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+
+        for dep in self.gdal_deps:
+            if "require" in dep and getattr(self.options, _option_name(dep)):
+                self.requires(dep["require"])
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        if "fPIC" in self.options and self.options.fPIC:
+            cmake.definitions[
+                "GDAL_OBJECT_LIBRARIES_POSITION_INDEPENDENT_CODE"] = True
+
+        cmake.definitions["BUILD_JAVA_BINDINGS"] = False
+        cmake.definitions["BUILD_CSHARP_BINDINGS"] = False
+        cmake.definitions["BUILD_PYTHON_BINDINGS"] = False
+
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.definitions["GDAL_USE_ZLIB_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_JSONC_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_JPEG_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_JPEG12_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_TIFF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_GEOTIFF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_GIF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_PNG_INTERNAL"] = False
+
+        cmake.definitions["GDAL_USE_LERC_INTERNAL"] = True
+        cmake.definitions["GDAL_USE_SHAPELIB_INTERNAL"] = True
+
+        cmake.definitions["BUILD_APPS"] = self.options.tools
+        del cmake.definitions["CMAKE_MODULE_PATH"]
+
+        cmake.definitions["SQLite3_HAS_COLUMN_METADATA"] = \
+            self.options["sqlite3"].enable_column_metadata
```

Comment:
Seems odd this can be True/False but https://github.com/conan-io/conan-center-index/pull/11694/files#r933543968 sets it to true by default

Reply:
the validate method is the right approach. In gdal cmake code, there is:
```
if (SQLite3_FOUND)
  if (NOT DEFINED SQLite3_HAS_COLUMN_METADATA)
    message(FATAL_ERROR "missing SQLite3_HAS_COLUMN_METADATA")
  endif ()
```
we might as well enforce it in the conanfile. I will do that.

Reply:
done